### DISCUSSION TITLE
Hash for gmail one click unsubscribe headers

### DISCUSSION THREAD

**Discussion by k040711** [postalserver/postal#2995] [Help with using Postal] [answered]
Hi everyone,

I'm trying to set up the additional headers google is asking for as of 2024's new guidelines for one click unsubscribe headers but having trouble understanding how to implement this in Postal's email API send

google requires 2 additional header lines:
(from google's documentation)
Header 1:
List-Unsubscribe-Post: List-Unsubscribe=One-Click

Header 2:
List-Unsubscribe: <https://myautoresponder.com/unsubscribe/example>

In postal's documentation additional headers are to be included as **hashed** <- this is the part i'm having trouble with

Can someone post an example of how to include this with an api request?

here's an example of a typical api send request i send:

{
    "from": "Me <info@mydomain.com>",
    "to": [
        "thesubscriber@gmail.com"
    ],
    "subject": "test email",
    "html_body": "<p>testing lone click unsubscribe</p>"
}

when I tried to add an additional "headers" : to the json it gives me an error saying the value of headers is a string and its not hashed.

thank you.

**Comment by willpower232** [comment]
which programming language are you using? Are you using one of the libraries? https://docs.postalserver.io/developer/client-libraries

**Comment by k040711** [comment, marked as answer]
yeah i figured it out,

all its asking for is a json object with the headers, like this:

"headers" : {
 "list-unsubscribe-post":"list-unsubscribe-post"
}
etc...

works great now.
